src/modules/mod.rs

```diff
@@ -51,39 +51,48 @@ pub(crate) async fn check_account_exists(
     .into())
 }
 
-pub(crate) async fn check_and_get_history_pagination_params(
-    pool: &sqlx::Pool<sqlx::Postgres>,
-    pagination_params: types::query_params::HistoryPaginationParams,
-) -> crate::Result<types::query_params::HistoryPagination> {
-    types::query_params::check_limit(pagination_params.limit)?;
-    let pagination = types::query_params::Pagination::from(pagination_params);
-    // if pagination_params.after_block_height.is_some() && pagination_params.after_timestamp_nanos.is_some() {
-    //     return Err(errors::ErrorKind::InvalidInput(
-    //         "Both block_height and block_timestamp_nanos found. Please provide only one of values"
-    //             .to_string(),
-    //     )
-    //         .into());
-    // }
-    // TODO PHASE 2 take the block from pagination_params
-    let block = db_helpers::get_last_block(pool).await?;
-    Ok(types::query_params::HistoryPagination {
-        block_height: block.height,
-        block_timestamp: block.timestamp,
-        limit: pagination.limit,
+/// Validates pagination_params received from the user
+pub(crate) async fn checked_get_pagination_params(
```

Comment:
Would implementing `From` be a better approach here?

```rs
impl From<types::query_params::PaginationParams> for types::query_params::Pagination {}
```

Reply:
I validate the input here, setting up default values. I'm not sure `TryFrom` would be correct from an ideological point of view. I feel `From/TryFrom` should be used to cast X to Y. We don't cast here, we do slightly more complex things